### PR TITLE
Remove duplicates for code generated types

### PR DESCRIPTION
**Pull request by HarelM**
## Launch Checklist

In order to export some part of the generated code in the near future (version 2.0 hopefully) to allow proper typing the types that are exported that are related to layout and paint properties need have a different names.

I changed the code generation code to allow this and also fixed the import to support this change.
There's not much here besides renaming of types...

### CODE REVIEW

**Comment by HarelM** [comment]
@drwestco if you could review this it would be great as you have fixed it I think. Thanks to your effort this was a real simple task :-)

File: build/generate-style-spec.ts [hunk 1]
```diff
@@ -158,19 +158,17 @@ export type CompositeFunctionSpecification<T> =
     | { type: 'interval',    stops: Array<[{zoom: number, value: number}, T]>, property: string, default?: T }
     | { type: 'categorical', stops: Array<[{zoom: number, value: string | number | boolean}, T]>, property: string, default?: T };
 
-export type ExpressionSpecification = Array<unknown>;
```
Comment (drwestco) [inline build/generate-style-spec.ts:161]: Might be worth keeping ExpressionSpecification as an alias for Array<unknown> for clarity. The fact that it's an array because it's an expression is meaningful.

File: src/style-spec/types.ts [hunk 2]
```diff
@@ -42,19 +42,17 @@ export type CompositeFunctionSpecification<T> =
     | { type: 'interval',    stops: Array<[{zoom: number, value: number}, T]>, property: string, default?: T }
     | { type: 'categorical', stops: Array<[{zoom: number, value: string | number | boolean}, T]>, property: string, default?: T };
 
-export type ExpressionSpecification = Array<unknown>;
```
Comment (drwestco) [inline src/style-spec/types.ts:45]: Same here. Seeing that a PropertyValueSpecification may be an ExpressionSpecification improves the readability of the type.

File: src/util/image.ts [hunk 3]
```diff
@@ -7,7 +7,7 @@ export type Size = {
   height: number;
 };
 
-type Point = {
+type XyPoint = {
```
Comment (drwestco) [inline src/util/image.ts:10]: XyPoint is a little funny-sounding. Point2D maybe?

**Review by drwestco — APPROVED** [review]

File: build/generate-style-spec.ts (hunk 1 above)
Comment (HarelM) [reply build/generate-style-spec.ts:161]: Can you think of another name for this type? It already exists in a different place...

**Comment by HarelM** [comment]
@drwestco I've changed things according to review. Thanks!

File: src/types/non-typed-modules.d.ts [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import type Pbf from 'pbf';
+import type Point from '../util/point';
```
Comment (wipfli) [inline src/types/non-typed-modules.d.ts:2]: Why is this needed?

**Review by wipfli — APPROVED** [review]

File: src/types/non-typed-modules.d.ts (hunk 4 above)
Comment (HarelM) [reply src/types/non-typed-modules.d.ts:2]: It is defined in the types below so it is needed for strict check which I'm exploring...

**Comment by wipfli** [comment]
Just to double-check: Do did not have to change tests because this is typescript only, and not because we have missing test coverage, right?

**Comment by HarelM** [comment]
I didn't need to change the tests, only the types.
